connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/IncrementalCooperativeAssignor.java

```diff
@@ -0,0 +1,537 @@
+
+        log.debug("Incremental Connector assignments: {}", incrementalConnectorAssignments);
+        log.debug("Incremental Task assignments: {}", incrementalTaskAssignments);
+
+        coordinator.leaderState(new LeaderState(memberConfigs, connectorAssignments, taskAssignments));
+
+        Map<String, ConnectAssignment> assignments =
+                fillAssignments(memberConfigs.keySet(), Assignment.NO_ERROR, leaderId,
+                                memberConfigs.get(leaderId).url(), maxOffset, incrementalConnectorAssignments,
+                                incrementalTaskAssignments, toRevoke, delay);
+
+        previousAssignment = ConnectorsAndTasks.embed(
+                connectorAssignments.values().stream().flatMap(Collection::stream).collect(Collectors.toSet()),
+                taskAssignments.values().stream().flatMap(Collection::stream).collect(Collectors.toSet()));
+
+        for (ConnectorsAndTasks revoked : toRevoke.values()) {
+            previousAssignment.connectors().removeAll(revoked.connectors());
+            previousAssignment.tasks().removeAll(revoked.tasks());
+        }
+
+        // Depends on the previous assignment's collections being sets at the moment.
+        // TODO: make it independent
+        previousAssignment.connectors().addAll(lostAssignments.connectors());
+        previousAssignment.tasks().addAll(lostAssignments.tasks());
+
+        log.debug("Actual assignments: {}", assignments);
+        return serializeAssignments(assignments);
+    }
+
+    // Factoring out with specific arguments to fix checkstyle error on method length.
+    // TODO: revisit passed arguments
+    private void handleLostAssignments(ConnectorsAndTasks lostAssignments,
+                                       ConnectorsAndTasks newSubmissions,
+                                       List<WorkerLoad> completeWorkerAssignment) {
+        if (!lostAssignments.isEmpty()) {
+            final long now = System.currentTimeMillis();
+            log.debug("Found the following connectors and tasks missing from previous assignment: "
+                    + lostAssignments);
+
+            if (scheduledRebalance > 0 && now >= scheduledRebalance) {
+                // delayed rebalance expired and it's time to assign resources
+                if (!candidateWorkersForReassignment.isEmpty()) {
+                    // At the moment we pick the first one.
+                    // The set is non-empty so it has at least one element.
+                    String recoveredWorker = candidateWorkersForReassignment.iterator().next();
+                    WorkerLoad workerLoad = completeWorkerAssignment.stream()
+                            .filter(w -> recoveredWorker.equals(w.worker()))
+                            .findAny()
+                            .orElse(null);
+
+                    if (workerLoad != null) {
+                        lostAssignments.connectors().forEach(c -> workerLoad.assign(c));
```

Comment:
nit: a method reference can be used here e.g., `forEach(workerLoad::assign)`

Reply:
Done